### ISSUE TITLE
[Tracking] Deferred components compatibility with add2app

### ISSUE THREAD

**Issue by GaryQian** [flutter/flutter#83461]
Deferred components currently supports a pure flutter app deferring pure flutter code and assets.

We have since seen requests to extend this to deferring in add2app use cases as well as the ability to fully defer all of flutter.

This can be multiplexed into the following support levels:

- Building flutter module into deferred components + base component that is included at initial install (https://github.com/flutter/flutter/issues/82791)
- Building the entire flutter module as a single deferrable component (https://github.com/flutter/flutter/issues/81265)
- Building the entire flutter module as a deferrable base module including the engine+framework as well as deferrable sub-components ('fully supported')
- No support for deferred components in add2app (this is where we currently are).

One of the primary challenges is adapting deferred components to an otherwise fully developer managed android app structure. It is risky to make assumptions about how the developer's app is set up and even riskier still to recommend/directly make changes to an existing non-flutter app.

**Comment by xster** [comment]
> It is risky to make assumptions about how the developer's app is set up and even riskier still to recommend/directly make changes to an existing non-flutter app.

Yes very risky. I think we'll have to lean even harder at having the deliverable be basically documentation.

**Comment by blasten** [comment]
I agree this sounds like documentation first, then evaluate if it makes sense to provide specific APIs that remove boilerplate, etc...

**Comment by xster** [comment]
@GaryQian where are we at with this?

**Comment by liqiyin** [comment]
Any updates?
